### ISSUE TITLE
`--environ-var` flag does not work, if fakemachine is disabled

### ISSUE THREAD

**Issue by sylv-io** [go-debos/debos#254]
If the host does not have kvm access or the `disable-fakemachine` flag is set, it is not possible to define environment variables via `--environ-var=VARIABLE:VALUE`.

e.g.:

fakemachine:
```
$ debos --cpus=1 --environ-var=LC_ALL:C --environ-var=LANG:C --environ-var=TZ:UTC0 --environ-var=SOURCE_DATE_EPOCH:1618827608  debug.yaml
2021/04/19 10:31:51 ==== Debug ====                                                                                            
2021/04/19 10:31:51 echo $SOURCE_DATE_EPOCH | 1618827608       
2021/04/19 10:31:51 ==== Exit ====                                                                                             
2021/04/19 10:31:51 Action `Exit` failed at stage Run, error: exit status 1                                                    
Powering off.
```

without fakemachine:
```
$ debos --disable-fakemachine --cpus=1 --environ-var=LC_ALL:C --environ-var=LANG:C --environ-var=TZ:UTC0 --environ-var=SOURCE_DATE_EPOCH:1618827608  debug.yaml
2021/04/19 10:42:26 fakemachine not supported, running on the host!
2021/04/19 10:42:26 ==== Debug ====
2021/04/19 10:42:26 echo $SOURCE_DATE_EPOCH | 
2021/04/19 10:42:26 ==== Exit ====
```

**Comment by sylv-io** [comment]
Looks like this is not a bug. The flag does only setup the environment for fakemachine:
```Go
  158     var m *fakemachine.Machine 

[...]

  279         // Puts in a format that is compatible with output of os.Environ()
  280         if context.EnvironVars != nil {
  281             EnvironString := []string{}
  282             for k, v := range context.EnvironVars {
  283                 warnLocalhost(k, v)
  284                 EnvironString = append(EnvironString, fmt.Sprintf("%s=%s", k, v))
  285             }
  286             m.SetEnviron(EnvironString) // And save the resulting environ vars on m
  287         }
  ```

**Comment by sylv-io** [comment]
I see two options to avoid misunderstandings:

1. `--environ-var` also sets the variable on the host, so it works without fakemachine
2. Update the description of `--environ-var` (e.g. `Fakemachine environment variables (use -e VARIABLE:VALUE syntax)`)

Personally I prefer option 1. What do you think about it? I could also prepare PR for this change.

**Comment by tobsan** [comment]
> I see two options to avoid misunderstandings:
> 
>     1. `--environ-var` also sets the variable on the host, so it works without fakemachine
> 
>     2. Update the description of `--environ-var` (e.g. `Fakemachine environment variables (use -e VARIABLE:VALUE syntax)`)
> 
> 
> Personally I prefer option 1. What do you think about it? I could also prepare PR for this change.

I would also prefer option 1 here.

**Comment by obbardc** [comment]
> > I see two options to avoid misunderstandings:
> > ```
> > 1. `--environ-var` also sets the variable on the host, so it works without fakemachine
> > 
> > 2. Update the description of `--environ-var` (e.g. `Fakemachine environment variables (use -e VARIABLE:VALUE syntax)`)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Personally I prefer option 1. What do you think about it? I could also prepare PR for this change.
> 
> I would also prefer option 1 here.

I agree. I wonder what we should do if the environment vat is already set before running debos? Error out?
Also what kind of cleanup is needed?

**Comment by tobsan** [comment]
> > > I see two options to avoid misunderstandings:
> > > ```
> > > 1. `--environ-var` also sets the variable on the host, so it works without fakemachine
> > > 
> > > 2. Update the description of `--environ-var` (e.g. `Fakemachine environment variables (use -e VARIABLE:VALUE syntax)`)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Personally I prefer option 1. What do you think about it? I could also prepare PR for this change.
> > 
> > 
> > I would also prefer option 1 here.
> 
> I agree. I wonder what we should do if the environment vat is already set before running debos? Error out? Also what kind of cleanup is needed?

It would be reasonable that the env var is only set during execution of debos, so any existing variables are reset to their previous value when debos exits, and that any other `--environ-var` variables are removed from the env at that point.
